### DISCUSSION TITLE
Thoughts on adding a ULID struct?

### DISCUSSION THREAD

**Discussion by pdevito3** [dotnet/runtime#91161] [Ideas] [answered]
[Continuing this discussion](https://github.com/dotnet/csharplang/discussions/7495)

I came across [this article](https://shopify.engineering/building-resilient-payment-systems) by shopify and thought point 6 was interesting about [ULID](https://github.com/ulid/spec)s.

The short version is they were able to realize a 50 percent decrease in INSERT statement duration by switching from UUIDv4 to ULID for idempotency keys.

Now, most of the operations in the majority of applications are reads vs writes, but this seemed interesting and figured i'd float the idea. I think the closest I've seen in our ecosystem at the moment is [Cysharp/Ulid](https://github.com/Cysharp/Ulid). There's also [Newid](https://github.com/phatboyg/NewId) but that's a bit different.

**Comment by Szer** [comment, marked as answer]
There were a lot of heated discussions about it recently
https://github.com/dotnet/runtime/issues/88290
https://github.com/dotnet/runtime/issues/86798
https://github.com/dotnet/runtime/issues/86084

Regarding UILD, just my 5 cents, there is no point in adding non-IETF-specced ULID data format to BCL (read: to the end of the .NET life) as UUID v7 is in a [draft](https://datatracker.ietf.org/doc/draft-ietf-uuidrev-rfc4122bis/)

**Reply by pdevito3** [reply]
Ah my bad. I checked the other discussions before I posted but not these once I got redirected.
